java/client/src/main/java/glide/api/models/commands/FT/FTCreateOptions.java

```diff
@@ -0,0 +1,427 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.FT;
+
+import static glide.api.models.GlideString.gs;
+
+import glide.api.BaseClient;
+import glide.api.commands.servermodules.FT;
+import glide.api.models.GlideString;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.NonNull;
+
+/**
+ * Additional parameters for {@link FT#create(BaseClient, String, FieldInfo[], FTCreateOptions)}
+ * command.
+ */
+@Builder
+public class FTCreateOptions {
+    /** The index type. If not given a {@link IndexType#HASH} index is created. */
+    private final IndexType indexType;
+
+    /** A list of prefixes of index definitions. */
+    private final GlideString[] prefixes;
+
+    FTCreateOptions(IndexType indexType, GlideString[] prefixes) {
+        this.indexType = indexType;
+        this.prefixes = prefixes;
+    }
+
+    public static FTCreateOptionsBuilder builder() {
+        return new FTCreateOptionsBuilder();
+    }
+
+    public GlideString[] toArgs() {
+        var args = new ArrayList<GlideString>();
+        if (indexType != null) {
+            args.add(gs("ON"));
+            args.add(gs(indexType.toString()));
+        }
+        if (prefixes != null && prefixes.length > 0) {
+            args.add(gs("PREFIX"));
+            args.add(gs(Integer.toString(prefixes.length)));
+            args.addAll(List.of(prefixes));
+        }
+        return args.toArray(GlideString[]::new);
+    }
+
+    public static class FTCreateOptionsBuilder {
+        public FTCreateOptionsBuilder prefixes(String[] prefixes) {
+            this.prefixes = Stream.of(prefixes).map(GlideString::gs).toArray(GlideString[]::new);
+            return this;
+        }
+    }
+
+    /** Type of the index dataset. */
+    public enum IndexType {
+        /** Data stored in hashes, so field identifiers are field names within the hashes. */
+        HASH,
+        /** Data stored in JSONs, so field identifiers are JSON Path expressions. */
+        JSON
+    }
+
+    /**
+     * A vector search field. Could be one of the following:
+     *
+     * <ul>
+     *   <li>{@link NumericField}
+     *   <li>{@link TextField}
+     *   <li>{@link TagField}
+     *   <li>{@link VectorFieldHnsw}
+     *   <li>{@link VectorFieldFlat}
+     * </ul>
+     */
+    public interface Field {
```

Comment:
Quick question: Are all Field classes, FieldInfo class used anywhere in FTCreateOptions?

Reply:
`Field` used in `FieldInfo` and `FieldInfo` is an `FT.create` argument. Not used in options